src/Controller/EnMarche/DelegatedAccessController.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Controller\EnMarche;
+
+use App\Controller\CanaryControllerTrait;
+use App\Entity\MyTeam\DelegatedAccess;
+use App\Entity\MyTeam\DelegatedAccessEnum;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Symfony\Component\Routing\Annotation\Route;
+
+class DelegatedAccessController extends Controller
+{
+    use CanaryControllerTrait;
+
+    /**
+     * @Route("/espace-partage/{uuid}", name="access_delegation_set")
+     */
+    public function delegatedSpace(DelegatedAccess $delegatedAccess, SessionInterface $session)
+    {
+        $this->disableInProduction();
+
+        $session->set(DelegatedAccess::ATTRIBUTE_KEY, $delegatedAccess->getUuid()->toString());
+
+        $routes = DelegatedAccessEnum::getFirstRoutesForType($delegatedAccess->getType());
+
+        return $this->redirectToRoute($routes[$delegatedAccess->getAccesses()[0]]);
```

Comment:
how do you handle the empty `accesses` value? `getAccesses()[0]` will throw an exception?

Reply:
this controller is not and must not be called when the delegated accesses does not have any accesses in it. But, I can add a check and throw an exception if this is the case